### PR TITLE
Correctly apply current-menu-ancestor class in Nav block

### LINKED ISSUE #50069: Navigation block: Current menu item and ancestor class inconsistent application
### Description

In Navigation block the `current-menu-item` class is being applied on the menu `li` tag. However, the `current-menu-ancestor` class is being applied on `li > a` tag.

Please fix the application of `current-menu-ancestor` class and apply it onto the parent `li` tag, not the anchor tag.

### Step-by-step reproduction instructions

Set a Navigation menu block with submenus and visit a page from the submenu. Inspect the classes applied on the Navigation block list items.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

WordPress 6.2, Gutenberg 15.6.2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by getdave**
## What?

Applies the `current-menu-ancestor` class to the `<li>` not the `<a>`

Closes https://github.com/WordPress/gutenberg/issues/50069

## Why?

Consistency with `current-menu-item` which is applied to the `<li>`.

## How?

Applies the `current-menu-ancestor` class to the `<li>` not the `<a>`

## Testing Instructions

You should be able to test this via Playground using the following link

https://playground.wordpress.net/?gutenberg-pr=67169

Note you can install any publicly available Theme for testing using the format:

```
https://playground.wordpress.net/?gutenberg-pr=67169&theme={theme-slug-here}
```

On your WordPress test site you can then...

- Create Navigation menu
- Add a Page
- Add a submenu under that page and add some links.
- Save
- View front of site
- Inspect HTML of top level link
- See class is on `li` not `a`.

### CODE REVIEW

**Comment by getdave** [comment]
I'm guessing this class isn't applied in the Editor because there is no concept of this there right?

**Comment by MaggieCabrera** [comment]
> I'm guessing this class isn't applied in the Editor because there is no concept of this there right?

that sounds about right, it would be hard to map

File: packages/block-library/src/navigation-submenu/index.php
Comment (MaggieCabrera) [inline packages/block-library/src/navigation-submenu/index.php:225]: I wonder how many websites are targeting the "wrong" selector...

Comment (getdave) [reply packages/block-library/src/navigation-submenu/index.php:225]: Not sure. `class` is not a first party API though but as we know we need to acknowledge the impact on existing Themes.

How has this been handled in the past? It's not like we can throw an error if someone is using that class.

Comment (MaggieCabrera) [reply packages/block-library/src/navigation-submenu/index.php:225]: I think we just consider the risk and go with it. I can't remember when we've done this recently. Maybe a dev note?

Comment (MaggieCabrera) [reply packages/block-library/src/navigation-submenu/index.php:225]: @WordPress/block-themers anyone remember an instance of class changing like this?

**Comment by MaggieCabrera** [comment]
I'm ok tentatively approving this if we get some more feedback from themers. If this only affect block themes, I've done a search on a dump I have of block themes in the themes repo and I don't see many themes that would be affected by this change

File: packages/block-library/src/navigation-submenu/index.php
Comment (carolinan) [reply packages/block-library/src/navigation-submenu/index.php:225]: See https://github.com/WordPress/gutenberg/issues/67134

Comment (mrwweb) [reply packages/block-library/src/navigation-submenu/index.php:225]: If I read this right that the `.wp-block-navigation-item__content` class will be removed by this change, it looks like it's [used by a considerable number of themes](https://www.wpdirectory.net/search/01JD89RMRY58KJ6QCFJGKAPHGP) and a few plugins too.

Comment (juanfra) [reply packages/block-library/src/navigation-submenu/index.php:225]: @mrwweb it looks like that class is only used to iterate there and to determine where to place the `current-menu-ancestor` class. So, this specific change doesn't mean that the `wp-block-navigation-item__content` class will be removed.

> anyone remember an instance of class changing like this?

I don't.

**Review by juanfra — COMMENTED** [review]
This is testing well for me! I really like that it brings parity with how things work in classic themes. 

My fear is also that we don't know how many sites could be targeting the wrong selector.   It looks like there's only a handful of [themes](https://www.wpdirectory.net/search/01JD9M1904H6WQF28VKW4S97YF) doing this in the repository, though.

Screenshot 2024-11-22 at 10 27 17

**Comment by MaggieCabrera** [comment]
> This is testing well for me! I really like that it brings parity with how things work in classic themes.
> 
> My fear is also that we don't know how many sites could be targeting the wrong selector. It looks like there's only a handful of [themes](https://www.wpdirectory.net/search/01JD9M1904H6WQF28VKW4S97YF) doing this in the repository, though.

Yeah, I haven't been able to find much more than that myself, I think this is a low risk change.

**Comment by MaggieCabrera** [comment]
Ugh, sorry I misclicked there :(

**Comment by getdave** [comment]
@bph Would it be worth dropping this in the Outreach channel in Slack for feedback?

**Comment by bph** [comment]
Absolutely!  You  can also ping the @WordPress/outreach group here on GitHub.

**Comment by getdave** [comment]
I'll leave this open a few more days to allow for further feedback.

Otherwise I think we just need an approving review (with the caveat for the approver that it's agreed that it's not 100% possible to know if this will have some negative side effects).

**Review by draganescu — APPROVED** [review]
The problem is valid, the code makes sense.

**Comment by draganescu** [comment]
I approved, although I do wonder about it a bit in terms of removing the class from the anchor, what is wrong with:

- do not remove the class on the anchor
- add a new class to the list item `current-menu-item-ancestor`

**Review by MaggieCabrera — APPROVED** [review]
Given how early in the release we are and how little the impact seems to be when inspecting the code for themes in the directory, I think this is good to land as is to get it out and get feedback

**Comment by getdave** [comment]
> I approved, although I do wonder about it a bit in terms of removing the class from the anchor, what is wrong with:
> 
> * do not remove the class on the anchor
> * add a new class to the list item `current-menu-item-ancestor`

I don't think we should double up. It could be very confusing. Either move it or leave it "as is".

**Comment by getdave** [comment]
I'm going to merge this one as soon as I can get the PHP unit tests to run correctly.

**Comment by juanfra** [comment]
@getdave I re-ran the failed tests, and they’re all passing now 🚀

**Comment by tyrann0us** [comment]
I apologize for commenting on a merged PR and for drifting slightly off-topic, but I would like to highlight an IMO important point:

> I'm ok tentatively approving this if we get some more feedback from themers. If this only affect block themes, I've done a search on a dump I have of block themes in the themes repo and I don't see many themes that would be affected by this change

and 

> Given how early in the release we are and how little the impact seems to be when inspecting the code for themes in the directory, I think this is good to land as is to get it out and get feedback

Thank you for taking the time to review the publicly available block-theme repository; that analysis is valuable. Nevertheless, the data set in the .org directory covers only a fraction of the ecosystem. Many, perhaps most, production themes are private, commercially licensed. This is especially true in enterprise environments. Those installations rarely surface in public scans, yet even minor markup changes can impose significant maintenance costs.

In light of this, I believe the block editor should adopt a consistent policy for changes that relocate or rename markup or CSS hooks. Ideally, new behavior would be introduced in parallel with the legacy behavior for e.g., a full release cycle, accompanied by a developer and/or an announcement in the Field Guide. This grace period gives theme authors adequate time to adapt. Afterward, the legacy path can be removed.

For this specific change, adding `current-menu-ancestor` to the `<li>` while retaining it on the `<a>` for one cycle would have preserved backward compatibility, avoided breaking existing selectors, and still signaled the preferred location going forward. A supplementary Theme Check rule that warns if `.current-menu-ancestor` appears on `<a>` tags could have been introduced, further easing migration.

Thank you! 🙇🏽‍♂️

**Pull request merged**
